show/main.py

```diff
@@ -129,6 +170,110 @@ def run_command(command, display_cmd=False):
     if rc != 0:
         sys.exit(rc)
 
+
+def get_interface_mode():
+    mode = os.getenv('SONIC_CLI_IFACE_MODE')
+    if mode is None:
+        mode = "default"
+    return mode
+
+
+# Global Alias object
+_alias = InterfaceAliasConverter()
+
+
+def run_command_in_alias_mode(command, linux_command=False):
+    """Run command and replace all instances of SONiC interface names
+       in output with vendor-sepecific interface aliases.
+    """
+
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+
+    while True:
+        output = process.stdout.readline()
+        if output == '' and process.poll() is not None:
+            break
+        if output:
+            if linux_command:
```

Comment:
Is there a way to make Linux commands the default (i.e., "else") case here, such that if the command doesn't match any of the if/elif cases, it is treated as a Linux command? This would eliminate the need for the `linux_command` parameter.

Reply:
I like the idea. But, without the linux_command it triggers lot of confusion in replacing interfaces from defautl to alias Currently we are using different substitution methods for linux commands and sonic utilities. Without  this differentiation we are forced to club the logic. This is an overhead to take care all possibilities.

Reply:
Could you explain what you mean by "we are forced to club the logic"? My thought is that if the command doesn't match any of the specific `if command == ...` clauses, it would fall through into a "default" case, and this case would handle all Linux commands.

Reply:
Joe, the linux/sonic commands use 2 different logics in replacing the interface names. I tested the else style of approach what you suggested. I moved the sonic_utitilies replacement logic inside a generic function "print_output_in_alias_mode"  and included in most of the  "if" statements.  
```
if command == "portstat":
               """Show interface counters"""
               if output.startswith("IFACE"):
                   output = output.replace("IFACE", "IFACE".rjust(
                                           _interface.vendor_name_max_length))
               print_output_in_alias_mode(output, index)<----

            elif command == "pfcstat":
               """Show pfc counters"""
               if output.startswith("Port Tx"):
                   output = output.replace("Port Tx", "Port Tx".rjust(
                                           _interface.vendor_name_max_length))

               elif output.startswith("Port Rx"):
                   output = output.replace("Port Rx", "Port Rx".rjust(
                                           _interface.vendor_name_max_length))
               print_output_in_alias_mode(output, index)<----
```
Since, sonic_utitilies replacement logic is safely binded inside the if statements now it is free to introduce else for linux_utitlites commands. This approach works fine. 

I agree with checking alias mode inside the run_command. Will raise pull request post testing.